cmd/lakectl/cmd/plugin_list.go

```diff
@@ -0,0 +1,143 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/spf13/cobra"
+)
+
+const pluginListTemplate = `The following lakectl-compatible plugins are available:
+
+{{- range .Plugins }}
+  {{- if or .Verbose .Warning }}
+  {{ .Path }}
+  {{- else }}
+  {{ .Name }}
+  {{- end }}
+  {{- if .Warning }}
+    - {{ "warning:" | yellow }} {{ .Warning }}
+  {{- end }}
+{{- end }}
+`
+
+type pluginInfo struct {
+	Name    string
+	Path    string
+	Warning string
+	Verbose bool
+}
+
+var pluginListCmd = &cobra.Command{
+	Use:   "list",
+	Short: "List available lakectl plugins",
+	Long:  `Scans the PATH for executables named "lakectl-*" and lists the detected plugins.`,
+	Run: func(cmd *cobra.Command, args []string) {
+		verbose, _ := cmd.Flags().GetBool("verbose")
+		envPathDirs := filepath.SplitList(os.Getenv("PATH"))
+
+		// Filter out duplicates from envPathDirs
+		var pathDirs []string
+		seen := make(map[string]struct{})
+		for _, dir := range envPathDirs {
+			if dir == "" {
+				// Skip empty entries in PATH
+				continue
+			}
+			if _, ok := seen[dir]; !ok {
+				seen[dir] = struct{}{}
+				pathDirs = append(pathDirs, dir)
+			}
+		}
+
+		// Iterate through pathDirs to build the final list with correct overshadowing
+		pluginsOutput := make([]pluginInfo, 0)
+		foundEffectivePlugins := make(map[string]string) // plugin name -> path of chosen executable
+
+		for _, dir := range pathDirs {
+			files, err := os.ReadDir(dir)
+			if err != nil {
+				// Silently ignore errors from non-existent or unreadable directories in PATH
+				continue
+			}
+
+			for _, file := range files {
+				fileName := file.Name()
+				if !strings.HasPrefix(fileName, "lakectl-") {
+					continue
+				}
+				pluginName := strings.TrimPrefix(fileName, "lakectl-")
+				if pluginName == "" {
+					continue
+				}
+
+				fullPath := filepath.Join(dir, fileName)
+				absPath, err := filepath.Abs(fullPath)
+				if err != nil {
+					// Silently ignore errors from file paths that cannot be resolved
```

Comment:
Report this, it should never happen and will indicate some weirdness if it dies.

Reply:
print warning in verbose mode.